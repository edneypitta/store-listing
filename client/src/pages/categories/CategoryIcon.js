import React from 'react'
import { isOpenNow } from '../../selectors/schedule'

export const getCategoryIcon = category => {
  const anyStoreOpen = category.stores
    .map(store => store.schedule)
    .some(isOpenNow)

  return anyStoreOpen ?
    category.openIcon :
    category.sleepIcon
}

const CategoryIcon = ({ category }) => {
  return (
    <div className='icon' title='Click to see stores'>
      <img src={getCategoryIcon(category)} alt='category icon' />
    </div>
  )
}

export default CategoryIcon
