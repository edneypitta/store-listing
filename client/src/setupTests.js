import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

const RealDate = Date
export const mockDate = isoDate =>
  global.Date = class extends RealDate {
    constructor() {
      return new RealDate(isoDate)
    }
  }