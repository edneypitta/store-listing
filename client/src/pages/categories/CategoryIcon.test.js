import React from 'react'
import { render } from 'enzyme'
import { mockDate } from '../../setupTests'
import CategoryIcon from './CategoryIcon'

describe('CategoryIcon', () => {
  beforeEach(() => {
    mockDate('2019-03-05T12:00:00.977Z')
  })

  it('renders without crashing', () => {
    const category = {
      stores: [{ schedule: [{ day: 1, open: '11:00', close: '13:00' }] }]
    }
    const component = render(<CategoryIcon category={category} />)

    expect(component).toMatchSnapshot()
  })

  it('renders with open icon', () => {
    const category = {
      openIcon: 'openIcon',
      stores: [{ schedule: [{ day: 1, open: '11:00', close: '13:00' }] }]
    }
    const component = render(<CategoryIcon category={category} />)

    expect(component).toMatchSnapshot()
  })

  it('renders with sleep icon', () => {
    const category = { sleepIcon: 'sleepIcon', stores: [] }
    const component = render(<CategoryIcon category={category} />)

    expect(component).toMatchSnapshot()
  })
})

