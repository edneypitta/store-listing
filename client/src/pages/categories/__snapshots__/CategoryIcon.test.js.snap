// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryIcon renders with open icon 1`] = `
<div
  class="icon"
  title="Click to see stores"
>
  <img
    alt="category icon"
    src="openIcon"
  />
</div>
`;

exports[`CategoryIcon renders with sleep icon 1`] = `
<div
  class="icon"
  title="Click to see stores"
>
  <img
    alt="category icon"
    src="sleepIcon"
  />
</div>
`;

exports[`CategoryIcon renders without crashing 1`] = `
<div
  class="icon"
  title="Click to see stores"
>
  <img
    alt="category icon"
  />
</div>
`;
